brave/brave-core
# Reland fix for Android keychecker crash

Resolves https://github.com/brave/brave-browser/issues/28890

Original issue: https://github.com/brave/brave-browser/issues/28831.

This is re-land of https://github.com/brave/brave-core/pull/17431 . Original PR broken Nightly Release AAB build. The problem was in wrong dependencies. 

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Build Nightly Release x86 AAB
```
npm run create_dist -- Release --channel=nightly --target_os=android --target_arch=x86 --target_android_base=mono --target_android_output_format=aab
```
2. Install and ensure it runs well
```
java -jar ./src/third_party/android_build_tools/bundletool/bundletool.jar            \
build-apks --connected-device                                                        \
--bundle=./src/out/android_Release_x86/apks/BraveMonox86.aab                         \
--output=./src/out/android_Release_x86/apks/BraveMonox86.apks                        \
--adb=./src/third_party/android_sdk/public/platform-tools/adb                        \
--aapt2=./src/third_party/android_build_tools/aapt2/aapt2                            \


java -jar ./src/third_party/android_build_tools/bundletool/bundletool.jar            \
install-apks                                                                         \
--apks=./src/out/android_Release_x86/apks/BraveMonox86.apks                          \
--adb=./src/third_party/android_sdk/public/platform-tools/adb

```
